src/core/dep-graph.ts

```diff
@@ -124,13 +124,21 @@ class DepGraphImpl implements types.DepGraphInternal {
     return this._hasCycles;
   }
 
-  public pkgPathsToRoot(pkg: types.Pkg): types.PkgInfo[][] {
+  public pkgPathsToRoot(
+    pkg: types.Pkg,
+    opts?: { limit?: number },
+  ): types.PkgInfo[][] {
     const pathsToRoot: types.PkgInfo[][] = [];
     for (const nodeId of this.getPkgNodeIds(pkg)) {
-      const pathsFromNodeToRoot = this.pathsFromNodeToRoot(nodeId);
+      const pathsFromNodeToRoot = this.pathsFromNodeToRoot(nodeId, [], {
+        limit: opts?.limit,
+      });
       for (const path of pathsFromNodeToRoot) {
         pathsToRoot.push(path);
       }
+      if (opts?.limit && pathsToRoot.length > opts.limit) {
```

Comment:
Given all of this is so hot, I'd move the optional chaining outside of the loop.
```suggestion
      if (limit > 0 && pathsToRoot.length > opts.limit) {
```
With something like this at the start of the method:
```
const limit = opts?.limit || 0;
```

Reply:
This was marked as resolved, but is still using `opts?.limit`

Reply:
Resolved^2